File: lib/models/paymentQRKey.dart
import 'dart:convert';

class PaymentQRKey {
  final String user_id;
  final String id;
  final String qrStr;

  PaymentQRKey({
    required this.user_id,
    required this.id,
    required this.qrStr,
  });
  
  Map<String, dynamic> toJson() {
    return {
      'user_id': user_id,
      'id': id,
      'qrStr': qrStr,
    };
  }

  factory PaymentQRKey.fromJson(Map<String, dynamic> json) {
    return PaymentQRKey(
      id: json['_id'],
      user_id: json['user_id'],
      qrStr: json['qrStr'],
    );
  }
}

File: server/models/paymentQRKey.js
const mongoose = require("mongoose");

const paymentQRKeySchema = mongoose.Schema({
  user_id: {
    type: String,
    required: true,
    trim: true,
  },
  qrStr: {
    type: String,
  }
});

// to convert a schema to a model
const PaymentQRKey = mongoose.model("PaymentQRKey", paymentQRKeySchema);
module.exports = PaymentQRKey;
